patch-diff-report-tool/src/main/resources/header.template

```diff
@@ -91,23 +91,23 @@
 							<td id="filesBase" th:text=${statistics.fileNumBase}> fileNumBase </td>
 							<td id="totalBase" th:text=${statistics.totalNumBase}> totalNumBase </td>
 							<th:block th:each="name : ${statistics.severityNames}">
-								<td th:id="${name} + @{SeverityNumBase}" th:text="${statistics.severityNumBase.get(name) != null} ? ${statistics.severityNumBase.get(name)} : '0'"> infoNumBase </td>
+								<td th:id="${name} + @{SeverityNumBase}" th:text="${statistics.severityNumBase.get(name)} ?: '0'"> infoNumBase </td>
 							</th:block>
 						</tr>
 						<tr class="a">
 							<td>patch</td>
 							<td id="filesPatch" th:text=${statistics.fileNumPatch}> fileNumPatch </td>
-							<td id="totalPatch" th:text=${statistics.totalNumPatch}> totalNumPatch </td>
+							<td id="totalPatch" th:text="${statistics.totalNumPatch} + (${statistics.uniqueMessagesBase > 0 || statistics.uniqueMessagesPatch > 0} ? (' (' + (${statistics.uniqueMessagesBase > 0} ? |${statistics.uniqueMessagesBase} removed| + (${statistics.uniqueMessagesPatch > 0} ? ', ' : '') : '') + (${statistics.uniqueMessagesPatch > 0} ? |${statistics.uniqueMessagesPatch} added| : '') + ')') : '')"> totalNumPatch </td>
 							<th:block th:each="name : ${statistics.severityNames}">
-								<td th:id="${name} + @{SeverityNumPatch}" th:text="${statistics.severityNumPatch.get(name) != null} ? ${statistics.severityNumPatch.get(name)} : '0'"> infoNumBase </td>
+								<td th:id="${name} + @{SeverityNumPatch}" th:text="(${statistics.severityNumPatch.get(name)} ?: '0') + (${statistics.severityNumRemoved.get(name) || statistics.severityNumAdded.get(name)} ? (' (' + (${statistics.severityNumRemoved.get(name)} ? |${statistics.severityNumRemoved.get(name)} removed| + (${statistics.severityNumAdded.get(name)} ? ', ' : '') : '') + (${statistics.severityNumAdded.get(name)} ? |${statistics.severityNumAdded.get(name)} added| : '') + ')') : '')"> infoNumBase </td>
```

Comment:
Can we move these complex logics to some class so it can look nicer and easier to read?

Reply:
all logic should be moved to java code. Template just show well prepared string values

Reply:
done